src/Uno.UI/UI/Xaml/DependencyObjectStore.Binder.cs

```diff
@@ -562,9 +562,17 @@ internal void SetBindingValue(DependencyPropertyDetails propertyDetails, object
 			return null;
 		}
 
-		private void OnDependencyPropertyChanged(DependencyPropertyDetails propertyDetails, DependencyPropertyChangedEventArgs args)
+		private void OnDependencyPropertyChanged(DependencyPropertyDetails propertyDetails, DependencyPropertyChangedEventArgs args, bool isUnsetValue)
 		{
-			SetBindingValue(propertyDetails, args.NewValue);
+			if (isUnsetValue)
+			{
+				propertyDetails.ClearBinding();
+			}
+			else
+			{
+				SetBindingValue(propertyDetails, args.NewValue);
+			}
```

Comment:
I'm not too sure about removing `SetBindingValue` when `UnsetValue` is used. Is this what UWP/WinUI does?

Reply:
@jeromelaban My assumption was based on running the following snippet on UWP:

```csharp
			var textBox = new TextBox();
			textBox.SetBinding(TextBox.TextProperty, new Binding());
			var bindingExpr1 = textBox.GetBindingExpression(TextBox.TextProperty);
			textBox.SetValue(TextBox.TextProperty, DependencyProperty.UnsetValue);
			var bindingExpr2 = textBox.GetBindingExpression(TextBox.TextProperty);
```

When running the above code in UWP, I see that `bindingExpr1` is non-null and `bindingExpr2` is null.

That's why I assumed that setting DP to `DependencyProperty.UnsetValue` should clear bindings.

Reply:
Indeed it does clear the binding, but indeed, as we're removing the bindings, there's no need to propagate the value. That being said, I'm wondering if there's any sort of propagation that happens to the source anyways. Can you check if there's any callback happening on a `INotifyPropertyChanged` binding on source ?

Reply:
> Can you check if there's any callback happening on a `INotifyPropertyChanged` binding on source ?

I'm not exactly sure about the scenario you're describing. Do you mean something like this?

```csharp
    public sealed partial class MainPage2 : Page
    {
        public readonly Button MyButton;
        public readonly CheckBox MyCheckBox;
        public readonly Model MyModel;

        public class Model : INotifyPropertyChanged
        {
            private bool isChecked = true;

            public bool IsChecked
            {
                get => isChecked;
                set
                {
                    isChecked = value;
                    if (PropertyChanged != null)
                    {
                        PropertyChanged(this, new PropertyChangedEventArgs(nameof(IsChecked)));
                    }
                }
            }

            public event PropertyChangedEventHandler PropertyChanged;
        }

        public MainPage2()
        {
            MyCheckBox = new CheckBox();
            MyCheckBox.Unchecked += CheckBox_CheckedChanged;
            MyCheckBox.Checked += CheckBox_CheckedChanged;
            MyButton = new Button
            {
                Content = "Click me",
            };
            MyModel = new Model();
            MyModel.PropertyChanged += MyModel_PropertyChanged;
            Content = new StackPanel
            {
                Orientation = Orientation.Horizontal,
                Children =
                {
                    MyCheckBox.Bind(MyModel, nameof(Model.IsChecked)),
                    MyButton.OnClick((s, e) => MyCheckBox.ClearValue(CheckBox.IsCheckedProperty)),
				}
            };
        }

        private void MyModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
        {
            // Check if this is hit?
        }

        private void CheckBox_CheckedChanged(object sender, RoutedEventArgs e) { }
        
    }

    public static class Extensions
    {
        public static TElement Bind<TElement>(this TElement element, object source, string path = null, IValueConverter converter = null, object converterParameter = null, BindingMode mode = BindingMode.TwoWay) where TElement : FrameworkElement
        {
            element.SetBinding(CheckBox.IsCheckedProperty, new Binding
            {
                Converter = converter,
                ConverterParameter = converterParameter ?? element,
                Mode = mode,
                Path = string.IsNullOrEmpty(path) ? null : new PropertyPath(path),
                Source = source ?? element,
            });
            return element;
        }

        public static TElement OnClick<TElement>(this TElement element, RoutedEventHandler handler) where TElement : Button
        {
            element.Click += handler;
            return element;
        }
    }
```

If that's what you meant, then no, `MyModel_PropertyChanged` isn't hit (both UWP and this PR)